src/core/contracts/http-adapter.ts

```diff
@@ -1,8 +1,8 @@
-abstract class HttpAdapter {
```

Comment:
@ctsstc @mgred I am not fully certain if there is a difference in TS with abstract vs interfaces since ultimately they're both just getting you to adhere to a contract. Thoughts? Is interface the right way to go?

Reply:
I think `interface` is the right way to do it.

Reply:
Sweet that's what I wanted to know!

Reply:
I don't remember the exactly reason right now, but I was torn and actually flip flopped on it twice before finally re-landing back to interface. I think it's because you were trying to instantiate it [iirc] which you cannot do with abstract classes.